File: core/src/com/vitekkor/polytech/objects/Student.java
package com.vitekkor.polytech.objects;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.g2d.Animation;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.graphics.g2d.TextureAtlas;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.utils.Array;

import static com.badlogic.gdx.scenes.scene2d.actions.Actions.moveTo;
import static com.badlogic.gdx.scenes.scene2d.actions.Actions.sequence;

public class Student extends Actor {
    private Array<TextureAtlas.AtlasRegion> regions;
    private Animation animation;
    private float stateTime;
    private float positionX = 0f;
    private float positionY = 0f;
    private float myWidth;
    private float myHeight;
    private float lastTime = 0;
    private boolean jump;

    public Student(TextureAtlas textureAtlas) {
        regions = textureAtlas.getRegions();
        setPosition(20f, 480f);
        setSize(regions.get(0).getRegionWidth(), regions.get(0).getRegionHeight());
        setScale(2f);
        animation = new Animation(1 / 8f, regions);
    }

    public void draw(Batch batch, float parentAlpha) {
        super.draw(batch, parentAlpha);
        stateTime += Gdx.graphics.getDeltaTime();
        batch.draw((TextureRegion) animation.getKeyFrame(stateTime, true), positionX, positionY, myWidth, myHeight);

    }

    @Override
    public void setPosition(float x, float y) {
        positionX = x;
        positionY = y;
        positionChanged();
    }

    @Override
    public void setPosition(float x, float y, int alignment) {
        setPosition(x, y);
    }

    @Override
    public void setScale(float scaleXY) {
        setSize(myWidth * scaleXY, myHeight * scaleXY);
    }

    @Override
    public void setSize(float width, float height) {
        myHeight = height;
        myWidth = width;
    }

    @Override
    public float getX(int alignment) {
        return positionX;
    }

    @Override
    public float getY(int alignment) {
        return positionY;
    }

    private void setRegions(TextureAtlas newTextureAtlas) {
        regions = newTextureAtlas.getRegions();
    }

    public void jump() {
        if (positionY == 480)
            addAction(sequence(moveTo(positionX, positionY + 160f, 2 / 7f), moveTo(positionX, positionY, 2 / 7f)));
    }

    public void go(boolean right) {
        float x = positionX;
        if (right) x += 20;
        else x -= 20;
        addAction(moveTo(x, positionY, 1 / 7f));
    }
}

File: core/src/com/vitekkor/polytech/objects/Student.kt
package com.vitekkor.polytech.objects

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.math.Interpolation
import com.badlogic.gdx.math.Vector2
import com.badlogic.gdx.scenes.scene2d.Actor
import com.vitekkor.polytech.supportFiles.AssetsLoader
import com.vitekkor.polytech.supportFiles.Move


class Student(x: Float, y: Float, private var width: Int, private var height: Int) : Actor() {

    private var targetPosition = Vector2(x, y)
    private var runTime = 0f
    private var position = Vector2(x, y)
    private var jumpUp = false
    private var jumped = true
    private var jumpDown = false
    private var myMove = Move(1 / 60f, Interpolation.linear)

    init {
        myMove.duration = 1 / 60f
        myMove.interpolation = Interpolation.linear
        myMove.setActor(this)
    }

    fun update(delta: Float) {
        if (!jumpDown && !jumpUp && !jumped) {
            myMove.duration = 1 / 60f
            jumped = true
        }
        myMove.setPosition(targetPosition.x, targetPosition.y)
        if (myMove.act(delta)) {
            if (!jumpUp && jumpDown) {
                jumpDown = false
            }
            if (jumpUp) {
                jumpUp = false
                targetPosition = Vector2(targetPosition.x, targetPosition.y - 120f)
            }
            myMove.restart()
        }
    }

    fun jump() {
        if (jumped) {
            jumpUp = true
            jumpDown = true
            jumped = false
            targetPosition = Vector2(targetPosition.x, targetPosition.y + 120f)
            myMove.duration = 1 / 7f
            myMove.restart()
        }
    }

    fun go(right: Boolean) {
        targetPosition = if (right) Vector2(targetPosition.x + 5f, targetPosition.y) else Vector2(targetPosition.x - 5f, targetPosition.y)
    }

    override fun getX(): Float {
        return position.x
    }

    override fun getY(): Float {
        return position.y
    }

    override fun getX(alignment: Int): Float {
        return position.x
    }

    override fun getY(alignment: Int): Float {
        return position.y
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        super.draw(batch, parentAlpha)
        runTime += Gdx.graphics.deltaTime
        //update(runTime)
        //targetPosition = position
        batch.draw(AssetsLoader.studentAnimation!!.getKeyFrame(runTime, true) as TextureRegion, x, y, getWidth(), getHeight())
    }

    override fun getWidth(): Float {
        return width.toFloat()
    }

    override fun getHeight(): Float {
        return height.toFloat()
    }

    override fun setPosition(x: Float, y: Float, alignment: Int) {
        position.x = x
        position.y = y
    }

    override fun setPosition(x: Float, y: Float) {
        position.x = x
        position.y = y
    }
}